extensions/sql-database-projects/src/test/dialogs/createProjectFromDatabaseQuickpick.test.ts

```diff
@@ -0,0 +1,316 @@
+		// user doesn't choose any folder when prompted and exits the showOpenDialog
+		openDialogStub.onSecondCall().resolves(undefined);
+		// user chooses to browse for folder
+		quickPickStub.onCall(3).resolves((constants.browseEllipsisWithIcon) as any);
+		// user doesn't choose any folder when prompted and exits the showOpenDialog
+		openDialogStub.onSecondCall().resolves(undefined);
+		//user chooses to exit
+		quickPickStub.onCall(4).resolves(undefined);
+
+		const model = await createProjectFromDatabaseQuickpick.createNewProjectFromDatabaseWithQuickpick(mockConnectionInfo);
+
+		//verify showQuickPick exited with undefined, since project location wasn't selected (resolved to undefined)
+		should.equal(model, undefined);
+	});
+
+	it('Should exit when folder structure is not selected and folder is selected through browsing (test repeatedness for project location)', async function (): Promise<void> {
+		sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);	//set vscode mssql extension api
+
+		let dbList: string[] = constants.systemDbs;
+		dbList.push('OtherDatabase');
+		dbList.push('Database');
+		dbList.push('OtherDatabase2');
+
+		//user chooses connection and database
+		sinon.stub(testUtils.vscodeMssqlIExtension.object, 'connect').resolves('testConnectionURI');
+		sinon.stub(testUtils.vscodeMssqlIExtension.object, 'listDatabases').withArgs(sinon.match.any).resolves(dbList);
+		let quickPickStub = sinon.stub(vscode.window, 'showQuickPick').resolves('Database' as any);
+		//user chooses project name
+		sinon.stub(vscode.window, 'showInputBox').resolves('TestProject');
+		// user chooses to browse for folder
+		quickPickStub.onSecondCall().resolves((constants.browseEllipsisWithIcon) as any);
+		// user doesn't choose any folder when prompted and exits the showOpenDialog
+		let openDialogStub = sinon.stub(vscode.window, 'showOpenDialog').withArgs(sinon.match.any).resolves(undefined);
+		// user chooses to browse for folder again
+		quickPickStub.onThirdCall().resolves((constants.browseEllipsisWithIcon) as any);
+		// user chooses folder- stub out folder to be chosen (showOpenDialog)
+		openDialogStub.onSecondCall().resolves([vscode.Uri.file(projectFilePath)]);
+		//user chooses to exit when prompted for folder structure
+		quickPickStub.onCall(3).resolves(undefined);
+
+		const model = await createProjectFromDatabaseQuickpick.createNewProjectFromDatabaseWithQuickpick(mockConnectionInfo);
+
+		//verify showQuickPick exited with undefined, since folder structure wasn't selected (resolved to undefined)
+		should.equal(model, undefined);
+	});
+
+	it('Should exit when folder structure is not selected and existing folder/file location is selected', async function (): Promise<void> {
+		sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);	//set vscode mssql extension api
+
+		//create folder and project file
```

Comment:
there should already be a helper function that creates a test project in testUtils that can be reused here